File: src/AutoContext.VsCode/src/types/instructions-file-parsed-result.ts
import type { InstructionsFileParsedSpan } from './instructions-file-parsed-span.js';
import type { InstructionsFileParserDiagnostic } from './instructions-file-parser-diagnostic.js';

export interface InstructionsFileParsedFrontmatter {
    readonly name?: string;
    readonly description?: string;
    readonly applyTo?: string;
    readonly version?: string;
}

export interface InstructionsFileParsedResult {
    readonly frontmatter: InstructionsFileParsedFrontmatter;
    readonly instructions: readonly InstructionsFileParsedSpan[];
    readonly diagnostics: readonly InstructionsFileParserDiagnostic[];
}

File: src/AutoContext.Instructions.Parser/InstructionsFileParsedResult.cs
namespace AutoContext.Instructions.Parser;

/// <summary>
/// The complete structural parse of one instructions file: its parsed frontmatter
/// paired with its parsed body (the normalised body text, the section index, the
/// actionable rule bullets, and any bullet-tag diagnostics). This is the single
/// shape every consumer — the build-time catalogue and metadata generators and the
/// runtime engine — reads, so the markdown is parsed once.
/// </summary>
/// <param name="Frontmatter">The parsed leading YAML frontmatter.</param>
/// <param name="Body">The parsed body: normalised text plus its section, rule, and
/// diagnostic index.</param>
public sealed record InstructionsFileParsedResult(
    InstructionsFileFrontmatterParsedResult Frontmatter,
    InstructionsFileBodyParsedResult Body);
